components/GameDetailsView.jsx

```diff
@@ -35,10 +37,16 @@ export default function GameDetailsView({
       progress: newValue,
     };
     onEditGame(updatedGame);
+
+    if (newValue === 100) {
+      setShowConfetti(true);
+      setTimeout(() => setShowConfetti(false), 5000);
+    }
   }
```

Comment:
In your `Animation.jsx` you use the cleanup function of `useEffect` to remove the interval, which is very good.

The same needs to be done when working with a `setTimeout` and it could look like this:

```suggestion
        }
  }
  
  useEffect(() => {
    if (showConfetti) {
      const timeOutId = setTimeout(() => setShowConfetti(false), 5000);

      return () => clearTimeout(timeOutId);
    }
  }, [showConfetti]);
```

You could keep the if-statement inside you `handleStatusChange` to set the `showConfetti` state to `true` when the `newValue` equals 100.

Outside of the `handleStatusChange` but inside a `useEffect` you could have your time out logic that only gets executed if `showConfetti` is `true` and uses the cleanup function to remove the timeout afterwards.

This change won't have a visual effect on your application, but the code will be cleaner, since we remove the timeout, once it is no longer needed.